File: backend/internal/domain/health.go
package domain

// HealthScore represents the overall health assessment of a portfolio.
type HealthScore struct {
	PortfolioID  int64             `json:"portfolio_id"`
	OverallScore int               `json:"overall_score"` // 0-100
	Grade        string            `json:"grade"`         // A, B, C, D, F
	Components   []HealthComponent `json:"components"`
	Alerts       []HealthAlert     `json:"alerts"`
	TopPicks     []TopPick         `json:"top_picks"`
}

// HealthComponent represents a single scoring dimension.
type HealthComponent struct {
	Name        string  `json:"name"`
	Score       int     `json:"score"` // 0-100
	Weight      float64 `json:"weight"`
	Description string  `json:"description"`
}

// HealthAlert represents a warning or informational notice about the portfolio.
type HealthAlert struct {
	Severity string `json:"severity"` // "info", "warning", "critical"
	Title    string `json:"title"`
	Message  string `json:"message"`
	Ticker   string `json:"ticker,omitempty"`
}

// TopPick represents a stock recommendation.
type TopPick struct {
	Ticker         string  `json:"ticker"`
	Name           string  `json:"name"`
	Action         string  `json:"action"` // "BUY", "HOLD", "WATCH"
	Reason         string  `json:"reason"`
	MarginOfSafety float64 `json:"margin_of_safety"`
	QualityScore   int     `json:"quality_score"`
}

File: frontend/src/lib/api/health.ts
import { api } from './client';

export interface HealthScore {
  portfolio_id: number;
  overall_score: number;
  grade: string;
  components: HealthComponent[];
  alerts: HealthAlert[];
  top_picks: TopPick[];
}

export interface HealthComponent {
  name: string;
  score: number;
  weight: number;
  description: string;
}

export interface HealthAlert {
  severity: 'info' | 'warning' | 'critical';
  title: string;
  message: string;
  ticker?: string;
}

export interface TopPick {
  ticker: string;
  name: string;
  action: 'BUY' | 'HOLD' | 'WATCH';
  reason: string;
  margin_of_safety: number;
  quality_score: number;
}

export async function getHealthScore(portfolioId: number): Promise<HealthScore> {
  return api<HealthScore>(`/portfolios/${portfolioId}/health`);
}

export async function getAlerts(portfolioId: number): Promise<HealthAlert[]> {
  return api<HealthAlert[]>(`/portfolios/${portfolioId}/health/alerts`);
}

export async function getTopPicks(portfolioId: number): Promise<TopPick[]> {
  return api<TopPick[]>(`/portfolios/${portfolioId}/health/top-picks`);
}
